pallets/parachain-staking/src/lib.rs

```diff
@@ -575,20 +578,35 @@ pub mod pallet {
 		}
 	}
 
-	/// Stores outstanding delegation requests per collator.
+	/// Stores outstanding delegation requests per collator & delegator.
+	///
+	/// Each `(collator, delegator)` pair can have up to
+	/// `T::MaxScheduledRequestsPerDelegator` scheduled requests,
+	/// which are always interpreted and executed in FIFO order.
 	#[pallet::storage]
 	#[pallet::getter(fn delegation_scheduled_requests)]
-	pub(crate) type DelegationScheduledRequests<T: Config> = StorageMap<
+	pub(crate) type DelegationScheduledRequests<T: Config> = StorageDoubleMap<
 		_,
 		Blake2_128Concat,
 		T::AccountId,
+		Blake2_128Concat,
+		T::AccountId,
 		BoundedVec<
 			ScheduledRequest<T::AccountId, BalanceOf<T>>,
```

Comment:
`ScheduledRequest` has a redundant `delegator` field at this point. Can we remove it here or should we add a TODO for the future?